_posts/2025-04-16-tha-various-tiers-of-tiered-storage.md

```diff
@@ -0,0 +1,454 @@
+For example:
+
+```
+docker build -t quay.io/scholzj/kafka:0.45.0-tiered-storage-kafka-3.9.0 .
+docker push docker push quay.io/scholzj/kafka:0.45.0-tiered-storage-kafka-3.9.0
+```
+
+Next, we need to prepare the shared volume that we will use as the remote storage tier.
+We can do that by creating a new Persistent Volume Claim (PVC) that will use the `nfs` storage class to provision the NFS volume:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: tiered-storage-nfs
+spec:
+  accessModes:
+    - ReadWriteMany
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 1Ti
+  storageClassName: nfs
+```
+
+With the container image and volume ready, we can now deploy the Kafka cluster.
+In the `Kafka` CR, we have to configure the container image with the tiered storage plugin in the `.spec.kafka.image` field.
+And we have to also enable the tiered storage in `.spec.kafka.tieredStorage`.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+  labels:
+    app: my-cluster
+  annotations:
+    strimzi.io/node-pools: enabled
+    strimzi.io/kraft: enabled
+spec:
+  kafka:
+    # ...
+    image: quay.io/scholzj/kafka:0.45.0-tiered-storage-kafka-3.9.0
+    listeners:
+      - name: plain
+        port: 9092
+        type: internal
+        tls: false
+    config:
+      # Tiered storage tunning
+      remote.log.manager.task.interval.ms: 5000
+      # Delete segments ever 10 seconds
+      log.retention.check.interval.ms: 10000
+      # ...
+    tieredStorage:
+      type: custom
+      remoteStorageManager:
+        className: io.aiven.kafka.tieredstorage.RemoteStorageManager
+        classPath: /opt/kafka/tiered-storage-filesystem/*
+        config:
+          storage.backend.class: io.aiven.kafka.tieredstorage.storage.filesystem.FileSystemStorage
+          storage.root: /mnt/tiered-storage/
+          storage.overwrite.enabled: "true"
+          chunk.size: "4194304"
+  # ...
+```
+
+And in all `KafkaNodePool` resources with the `broker` role, we have to mount the NFS volume in `.spec.template` using the additional volumes feature.
+The `KafkaNodePool` resources that have only the `controller` role do not need the volume as the would not use tiered storage.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: broker
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  roles:
+    - broker
+  # ...
+  template:
+    pod:
+      volumes:
+        - name: tiered-storage
+          persistentVolumeClaim:
+            claimName: tiered-storage-nfs
+    kafkaContainer:
+      volumeMounts:
+        - name: tiered-storage
+          mountPath: /mnt/tiered-storage/
+```
+
+Once the Kafka cluster is ready, we can to create a Kafka topic named `tiered-storage-test`.
+Keep in mind that in order to use the tiered storage, you always have to enable in the topic as well:
```

Comment:
Keep in mind that in order to use the tiered storage, you ~~always~~ have to enable in the topic as well: